src/plugins/event-plugins/NavigationPlugin.ts

```diff
@@ -1,295 +1,108 @@
-            responseTime:
-                entryData.responseStart > 0
-                    ? entryData.responseEnd - entryData.responseStart
-                    : 0,
-
-            domInteractive: entryData.domInteractive,
-            domContentLoadedEventStart: entryData.domContentLoadedEventStart,
-            domContentLoaded:
-                entryData.domContentLoadedEventEnd -
-                entryData.domContentLoadedEventStart,
-            domComplete: entryData.domComplete,
-            domProcessingTime: entryData.loadEventStart - entryData.responseEnd,
-            loadEventStart: entryData.loadEventStart,
-            loadEventTime: entryData.loadEventEnd - entryData.loadEventStart,
-
-            duration: entryData.duration,
-
-            headerSize: entryData.transferSize - entryData.encodedBodySize,
-            transferSize: entryData.transferSize,
-            compressionRatio:
-                entryData.encodedBodySize > 0
-                    ? entryData.decodedBodySize / entryData.encodedBodySize
-                    : 0,
-            navigationTimingLevel: 2
-        };
 
-        if (this.context?.record) {
-            this.context.record(
-                PERFORMANCE_NAVIGATION_EVENT_TYPE,
-                eventDataNavigationTimingLevel2
-            );
-        }
+            // Record
+            const e = entry as PerformanceNavigationTiming;
+            this.context?.record(PERFORMANCE_NAVIGATION_EVENT_TYPE, {
+                name: this.context.config.recordResourceUrl
+                    ? e.name
+                    : undefined,
+                entryType: NAVIGATION,
+                initiatorType: NAVIGATION,
+                startTime: e.startTime,
+                duration: e.duration,
+                nextHopProtocol: e.nextHopProtocol,
+                workerStart: e.workerStart,
+                redirectStart: e.redirectStart,
+                redirectEnd: e.redirectEnd,
+                fetchStart: e.fetchStart,
+                domainLookupStart: e.domainLookupStart,
+                domainLookupEnd: e.domainLookupEnd,
+                connectStart: e.connectStart,
+                connectEnd: e.connectEnd,
+                secureConnectionStart: e.secureConnectionStart,
+                requestStart: e.requestStart,
+                responseStart: e.responseStart,
+                responseEnd: e.responseEnd,
+                transferSize: e.transferSize,
+                encodedBodySize: e.encodedBodySize,
+                decodedBodySize: e.decodedBodySize,
+                domComplete: e.domComplete,
+                domContentLoadedEventEnd: e.domContentLoadedEventEnd,
+                domContentLoadedEventStart: e.domContentLoadedEventStart,
+                domInteractive: e.domInteractive,
+                loadEventEnd: e.loadEventEnd,
+                loadEventStart: e.loadEventStart,
+                redirectCount: e.redirectCount,
+                type: e.type,
```

Comment:
question: similar question as https://github.com/aws-observability/aws-rum-web/pull/489#discussion_r1456644095 for the [`type`](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceNavigationTiming/type) field

Reply:
+1 https://github.com/aws-observability/aws-rum-web/pull/494#issuecomment-1896993094